community/rust/musl-support-dynamic-linking.patch

```diff
@@ -38,14 +36,21 @@ diff --git a/src/bootstrap/compile.rs b/src/bootstrap/compile.rs
 index 7c35151a6d..c83b6cc24c 100644
 --- a/src/bootstrap/compile.rs
 +++ b/src/bootstrap/compile.rs
-@@ -94,14 +94,17 @@ pub fn std_link(build: &Build,
-     t!(fs::create_dir_all(&libdir));
-     add_to_sysroot(&out_dir, &libdir);
+@@ -94,21 +94,24 @@ pub fn std_link(build: &Build,
+     let libdir = build.sysroot_libdir(target_compiler, target);
+     add_to_sysroot(&libdir, &libstd_stamp(build, compiler, target));
  
 -    if target.contains("musl") && !target.contains("mips") {
 +    if target.contains("musl") {
          copy_musl_third_party_objects(build, target, &libdir);
      }
+ 
+     if build.config.sanitizers && compiler.stage != 0 && target == "x86_64-apple-darwin" {
+         // The sanitizers are only built in stage1 or above, so the dylibs will
+         // be missing in stage0 and causes panic. See the `std()` function above
+         // for reason why the sanitizers are not built in stage0.
+         copy_apple_sanitizer_dylibs(&build.native_dir(target), "osx", &libdir);
+     }
```

Comment:
Why is it here? This is relevant only on macOS…?

Reply:
I just copy it from file to fix patch

Reply:
Eh, yeah, I misunderstood it. :)